[LimHongYao] iP

LimHongYao opened the pull request:
Renamed Duke
Added ToDoList functionality (add, mark, unmark, show list)
Given some personality

LimHongYao committed:
Level-0

LimHongYao committed:
Added echo
Changed Duke's Name

LimHongYao committed:
Added list functionality

LimHongYao committed:
Added ability to mark and unmark as done
Improved handling of commands by splitting user input
Changed to use Tasks classes

khooyourun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,75 @@
+        boolean exit = false;
+        while (!exit) {
+            String input = (in.nextLine()).trim();
+            String inputCMD, inputItem;
```
Can I check what does CMD mean? If it stands for command. I think it's better to just name it as inputCommand for readibilty.

khooyourun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,75 @@
+            String inputCMD, inputItem;
+            if (input.contains(" ")) {
+                inputCMD = input.split(" ", 2)[0];
+                inputItem = input.split(" ", 2)[1];
```
The numbers like 2, 0, 1 might be considered magic numbers that hinder readability.

khooyourun commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,42 @@
+        }
+    }
+    public static void markDone (int index) {
+        if (TaskList.get(index).getStatusIcon().equals(" ")) {
```
Is this if statement necessary? I think markDone would change the status icon to "X" independent of what is originally is

khooyourun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,75 @@
+                    inputItem = in.nextLine().trim();
+                }
+
+                ToDoList.markDone(Integer.parseInt(inputItem)-1);
+                System.out.println("Okay I've marked item " + inputItem + " as done:");
+                ToDoList.printItem(Integer.parseInt(inputItem)-1);
```
I like the single level of abstraction applied here. Makes things easy to follow

khooyourun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,75 @@
+        System.out.println("Hi it's Anna!\nWhat do you need to do?");
+        Scanner in = new Scanner(System.in);
+
+        boolean exit = false;
```
Preferably name boolean in such a way that implies it is boolean.

liuziyang020319 replied:
Maybe use isExit to replace exit can imply a boolean

khooyourun reviewed (commented):
Overall, the code is readable and the if statements are not deeply nested.

liuziyang020319 commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+
+public class ToDoList {
+    private static final ArrayList<Task> TaskList = new ArrayList<>(10);
```
Maybe V\variable names need to be in camelCase.

liuziyang020319 replied:
Maybe variable names need to be in camelCase.

liuziyang020319 replied:
Maybe the size of the ArrayList should be a magic number.

liuziyang020319 commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,42 @@
+
+public class ToDoList {
+    private static final ArrayList<Task> TaskList = new ArrayList<>(10);
+    private static int NumTasks = 0;
```
Maybe variable names must be in camelCase, same problem.

liuziyang020319 reviewed (commented):
Most the codes follow the coding standard

LimHongYao committed:
Added deadline, event, todo types for list objects

LimHongYao committed:
fixed coding standard issues

LimHongYao committed:
added bad input handling
added immediate prompt for missing information
reworked the terrible code in getStartEnd

LimHongYao committed:
Merge branch 'branch-Level-5'

jeyvia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+import java.util.Scanner;
+
 public class Duke {
+    static final int COMMAND_INDEX = 0;
```
Good job using constants for your numbers! You could extend that to magic strings as well!

jeyvia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+                    TaskList.viewList();
+                }
+                break;
+
```
Do avoid leaving the space between each `case` as per the Java Coding Standards

jeyvia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+    public String getTypeIcon() {
+        return "NULL";
+    }
+    public String taskTypeIcon() { return "[" + getTypeIcon() + "]";}
```
Even though these are simple 1 liners, please do still follow the proper style (aka Egyptian Style).

jeyvia commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private static final ArrayList<Task> TaskList = new ArrayList<>(10);
```
ArrayList is a type of dynamic array and there is no need to limit your `TaskList` to a size of 10 only!

jeyvia commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+            System.out.println(TaskList.get(i).getTask());
+        }
+    }
+    public static void markDone (int index) {
```
There should not be a space between `markDone` and `(`. Do change this for your other methods as well :)

jeyvia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {   //ok to leave as public?
```
Yes you need to leave it as public. But do remove such comments from your code as it goes against the java coding standard. 👍

LimHongYao committed:
added delete functionality to list

LimHongYao committed:
added tasklist export functionality
fixed minor presentation error in event

LimHongYao committed:
reverted desktop export path to relative /out path
introduced some constants
introduced file path printout in event of export failure

LimHongYao committed:
Merge branch 'branch-Level-6'

LimHongYao committed:
Merge branch 'branch-Level-7'

LimHongYao committed:
packaged files
removed redundanct "add" method in Duke.java

LimHongYao committed:
moved some methods to Ui class

LimHongYao committed:
rewrote and combined mark/unmark methos

LimHongYao committed:
reworked and combined mark method

LimHongYao committed:
Moved switch cases into OOP
Reformatted event and deadline displays

LimHongYao committed:
added search functionality

LimHongYao committed:
Merge pull request #1 from LimHongYao/branch-Level-9

added search functionality

LimHongYao committed:
added parser class

LimHongYao committed:
Merge pull request #2 from LimHongYao/branch-Level-9

added parser class

LimHongYao committed:
added load from text file functionality

LimHongYao committed:
Added javadoc documentation

LimHongYao committed:
added javadoc documentation

LimHongYao committed:
Merge pull request #3 from LimHongYao/branch-A-JavaDoc

added javadoc documentation

LimHongYao committed:
Updated basic user guide

LimHongYao committed:
Added UserGuide documentation in docs directory for website

LimHongYao committed:
fixed bug with exception handling
fixed formatting for Deadline objects

LimHongYao committed:
re-commit with missing DukeException file
